packages/calcite-components/src/utils/dom.ts

```diff
@@ -426,6 +426,68 @@ export function toAriaBoolean(value: boolean): string {
   return Boolean(value).toString();
 }
 
+/**
+ * This helper returns a string of textContent if the target `slot` element from the `onSlotchange` event has any text content.
+ *
+ * ```
+ * <slot onSlotchange={(event) => this.mySlotText = slotChangeGetTextContent(event)} />}
+ * ```
+ *
+ * @param {Event} event The event.
+ * @returns {string} The slots text.
+ */
+export function slotChangeGetTextContent(event: Event): string {
+  return slotChangeGetAssignedNodes(event)
+    .filter((node) => node.nodeType === Node.TEXT_NODE)
+    .map((node) => node.textContent)
+    .join("")
+    .trim();
+}
```

Comment:
@anveshmekala I think this utility could be used for...

https://github.com/Esri/calcite-design-system/blob/bb07831eb518d1d617b765e1f14279480c6051fc/packages/calcite-components/src/components/text-area/text-area.tsx#L406-L415

What do you think?

Reply:
I'm a little confused about the logic above though, it seems like the last textNode will be set as `this.value` and any previous nodes would be overwritten.

Reply:
@anveshmekala I think this breaks the logic:

```html
<calcite-text-area>
Hello
world
hi</calcite-text-area>
```

Reply:
https://codepen.io/driskull/pen/oNQREpy?editors=1111

It seems to work but could you double check?

Reply:
it works with multi line text including special characters.